lib/rouge/lexers/perl6pod.rb

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Pod6 < RegexLexer
+      title "Perl 6 Pod"
+      desc "Perl 6 Plain' Ol Documentation (perl6.org)"
+
+      tag 'perl6pod'
+      aliases 'pod6'
+
+      filenames '*.pod', '*.pod6'
+
+      formatting_tokens = {
+        "B" => Generic::Strong,
+        "C" => Generic::Output,
+        "E" => Literal::Number::Other,
+        "I" => Generic::Emph,
+        "K" => Generic::Inserted,
+        "L" => Other,
+        "N" => Comment::Special,
+        "T" => Generic::Output,
+        "U" => Generic,
+        "Z" => Comment,
+      }
+
+      state :root do
+        rule(/\=begin pod/m, Keyword, :pod6)
+        rule(/^#.*$/, Comment)
+        rule(/./m, Text)
+      end
+
+      state :pod6 do
+        rule(/\=end pod/m, Keyword, :pop!)
+
+        rule(/\=(?:NAME|AUTHOR|VERSION|TITLE|SUBTITLE)/, Keyword, :semantic)
+
+        rule(/\=begin code/, Keyword, :block_code)
+        rule(/\=begin input/, Keyword, :block_input)
+        rule(/\=begin output/, Keyword, :block_output)
+        rule(/\=begin/, Keyword, :block)
+
+        rule(/\=head(\d+)\s+/, Keyword, :head)
+        rule(/\=item(\d+)\s+/, Keyword, :item)
+        rule(/\=input/, Keyword, :input)
+        rule(/\=output/, Keyword, :output)
+        rule(/\=defn/, Keyword)
+
+        rule(/(BCEIKLNTUZ)<([^>]*)>/) do |m|
+          t = formatting_tokens[m[0][0]]
+          groups Punctuation::Indicator, t, Punctuation::Indicator
+        end
+
+        rule(/^(?:\t|\s{4,})/, Other, :code)
+
+        rule(/./m, Text)
+      end
+
+      state :semantic do
+        rule(/.+\n/, Name, :pop!)
```

Comment:
It's not usually the case that a name would include the newline. If you can't remove this without breaking the lexer, that suggests you need a rule that will match whitespace in the state that is usually below this one.

Reply:
Alternatively, you can have the newline as a separate rule that pops the state (as you do in the `:head` state).